emacs/caml.el

```diff
@@ -866,10 +866,10 @@ possible."
  (if (eq major-mode 'caml-mode)
      (let (skip bol beg end)
        (save-excursion
-         (set-buffer
-          (if (boundp 'compilation-last-buffer)
-              compilation-last-buffer   ;Emacs 19
-            "*compilation*"))           ;Emacs 18
+         (with-current-buffer
+             (if (boundp 'compilation-last-buffer)
+                 compilation-last-buffer   ;Emacs 19
+               "*compilation*"))           ;Emacs 18
```

Comment:
Indeed, see https://www.gnu.org/software/emacs/manual/html_node/elisp/Excursions.html for more information.